packages/interaction-tracking/src/InteractionTracking.js

```diff
@@ -0,0 +1,263 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {
+  enableInteractionTracking,
+  enableInteractionTrackingObserver,
+} from 'shared/ReactFeatureFlags';
+import now from './InteractionTrackingNow';
+
+export type Interaction = {|
+  __count?: number,
+  id: number,
+  name: string,
+  timestamp: number,
+|};
+
+export type Subscriber = {
+  // A new interaction has been created via the track() method.
+  onInteractionTracked: (interaction: Interaction) => void,
+
+  // All scheduled async work for an interaction has finished.
+  onInteractionScheduledWorkCompleted: (interaction: Interaction) => void,
+
+  // New async work has been scheduled for a set of interactions.
+  // When this work is later run, onWorkStarted/onWorkStopped will be called.
+  // A batch of async/yieldy work may be scheduled multiple times before completing.
+  // In that case, onWorkScheduled may be called more than once before onWorkStopped.
+  // Work is scheduled by a "thread" which is identified by a unique ID.
+  onWorkScheduled: (interactions: Set<Interaction>, threadID: number) => void,
+
+  // A batch of scheduled work has been cancelled.
+  // Work is done by a "thread" which is identified by a unique ID.
+  onWorkCancelled: (interactions: Set<Interaction>, threadID: number) => void,
```

Comment:
I hate to nit on this, but do we do `Canceled` or `Cancelled`? (Failed) TC39 cancellation proposal used [one l](https://github.com/tc39/proposal-cancellation/blob/master/stage0/README.md#cancellationtokencanceled), so does [.NET](https://msdn.microsoft.com/en-us/library/system.threading.tasks.task.iscanceled(v=vs.110).aspx). Java uses [two l](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html#isCancelled()).

DOM has an API called [`cancelable`](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable) so I think one "c" would be more consistent.